code/game/machinery/doors/airlock.dm

```diff
@@ -784,20 +786,27 @@
 /obj/machinery/door/airlock/Topic(href, href_list, var/nowindow = 0)
 	// If you add an if(..()) check you must first remove the var/nowindow parameter.
 	// Otherwise it will runtime with this kind of error: null.Topic()
-	if(!nowindow)
-		..()
-	if(!usr.canUseTopic(src) && !IsAdminGhost(usr))
-		return
-	add_fingerprint(usr)
-
-	if((in_range(src, usr) && isturf(loc)) && panel_open)
-		usr.set_machine(src)
+	if(href_list["remoteOpen"])
+		if(request_made && try_to_activate_door(usr))
+			request_made = FALSE
```

Comment:
There's a security issue here. Topic() calls can be spoofed, so you need to make sure that the PDA in question did in fact click the link. Way I would do it is to pass a ref to the PDA as an arg in the Topic() and store a ref to the door accepted as a var on the PDA. Then you can just test if that door == src.